21.1 release notes and upgrade instructions (#10500)
cockroachdb/docs: releases/v21.1.0.md

Recovery | **Re-validating Indexes During `RESTORE`** | Incremental backups created by v20.2.2 and prior v20.2.x releases or v20.1.4 and prior v20.1.x releases may include incomplete data for indexes that were in the process of being created. Therefore, when incremental backups taken by these versions are restored by v21.1.0, any indexes created during those incremental backups will be re-validated by [`RESTORE`](../v21.1/restore.html#restore-from-incremental-backups).
Operations | **Logging Improvements** | Logging events are now organized into channels and can be redirected to different outputs, and output formats are now configurable (including JSON). This enables network logging and automatic processing with third-party tools.

> Operations | **HTTP API Monitoring Endpoints** | TBD

Reviewer: ```suggestion
Operations | **Cluster API v2.0** | This [new API](../v21.1/cluster-api.html) for monitoring clusters and nodes builds on prior endpoints, offering a consistent REST interface that makes it easier to use with your choice of tooling. The API offers a streamlined authentication process and developer-friendly [reference documentation](../api/cluster/v2.html).
```

Core | **Virtual Computed Columns** | You can now create [virtual computed column](../v21.1/computed-columns.html), which are not stored on disk and are recomputed as the column data in the expression changes.
Core | **Renaming and Dropping User-Defined Types** | It's now possible to [rename a user-defined type](../v21.1/alter-type.html#rename-a-user-defined-type) and [drop a value in a user-defined type](../v21.1/alter-type.html#drop-a-value-in-a-user-defined-type).
Core | **Sequence Caching** | You can now create a sequence with the [`CACHE`](../v21.1/create-sequence.html#cache-sequence-values-in-memory) keyword to have the sequence cache its values in memory.  
Core | **Changing Sequence & View Ownership** | You can use the new `OWNER TO` parameter to change the owner of a [sequence](../v21.1/alter-sequence.html) or [view](../v21.1/alter-view.html).  
Core | **Show `CREATE` Statements for the Current Database** | You can now use [`SHOW CREATE ALL TABLES`](../v21.1/show-create.html#show-the-statements-needed-to-recreate-all-tables-views-and-sequences-in-the-current-database) to return the `CREATE` statements for all of the tables, views, and sequences in the current database.

> Core | **ORM Compatibility for Spatial** | [Spatial libraries for Hibernate, ActiveRecord, and Django](../v21.1/spatial-data.html#orm-compatibility) are now fully compatible with CockroachDB's spatial features.

Reviewer 2: To add: New empty, unimplemented `pg_catalog` tables see https://www.cockroachlabs.com/docs/v21.1/pg-catalog.html

### Database Operations

Version | Feature | Description
--------|---------|------------
Core | **Logging Improvements** | Log events are now organized into [logging channels](../v21.1/logging-overview.html) that address different [use cases](../v21.1/logging-use-cases.html). Logging channels can be freely mapped to log sinks and routed to destinations outside CockroachDB (including external log collectors). All logging aspects, including message format (e.g., JSON), are now [configurable](../v21.1/configure-logs.html) via YAML.
Core | **Cluster API v2.0** | This [new API](../v21.1/cluster-api.html) for monitoring clusters and nodes builds on prior endpoints, offering a consistent REST interface that's easier to use with your choice of tooling. The API offers a streamlined authentication process and developer-friendly [reference documentation](../api/cluster/v2.html).
Core | **OpenShift-certified<br>Kubernetes Operator** | You can now [deploy CockroachDB on the Red Hat OpenShift platform](../v21.1/deploy-cockroachdb-with-kubernetes-openshift.html) using the latest OpenShift-certified Kubernetes Operator.  

> Core | **Automated TLS Cert Setup** | Using the new `cockroach connect` command and a secure token you provide, you can now automate the creation and distribution of a cluster's CA (certificate authority) and node certificates. Note that this feature is an alpha release with core functionality that may not meet your requirements.

Reviewer: ```suggestion
Core | **Auto TLS certificate setup** | [Using the new `cockroach connect` command](../v21.1/auto-tls.html), you can now let CockroachDB handle the creation and distribution among nodes of a cluster's CA (certificate authority) and node certificates. Note that this feature is an alpha release with core functionality that may not meet your requirements.
```

* Renamed to Auto TLS to match what we're using in the docs.
* Doc PR now merged - link to docs included.
* Removed mention of the token; probably too much detail, distracting from the core point.

Core | **Built-in Timezone Library** | The CockroachDB binary now includes a copy of the [`tzdata` library](../v21.1/recommended-production-settings.html#dependencies), which is required by certain features that use time zone data. If CockroachDB can't find the `tzdata` library externally, it will now use this built-in copy.  

### Backward-incompatible changes

> Before [upgrading to CockroachDB v21.1](../v20.1/upgrade-cockroach-version.html), be sure to review the following backward-incompatible changes and adjust your deployment as necessary.

Reviewer: ```suggestion
Before [upgrading to CockroachDB v21.1](../v21.1/upgrade-cockroach-version.html), be sure to review the following backward-incompatible changes and adjust your deployment as necessary.
```
